logback-classic/src/test/java/ch/qos/logback/classic/pattern/ThrowableProxyConverterTest.java

```diff
@@ -206,18 +206,95 @@ public void shouldLimitTotalLinesExcludingSkipped() throws Exception {
         String[] lines = result.split(CoreConstants.LINE_SEPARATOR);
         assertThat(lines).hasSize(3 + 1);
     }
+    
+    @Test
+    public void nullMessage() {
+        /*
+        Ensures that a null message is not sent to the output.
+        */
+        Exception t = new Exception();
+        verify(t);
+    }
+
+    @Test
+    public void circular() {
+        Exception w = null;
+        try {
+            someMethod();
+        } catch (Exception e) {
+            w = new Exception("Wrapper", e);
+            e.initCause(w);
+        }
+        verify(w);
+    }
+
+    @Test
+    public void suppressedCircular() throws Exception {
+        assumeTrue(TestHelper.suppressedSupported()); // only execute on Java 7, would work anyway but doesn't make
+           
+        Exception ex = null;
+        try {
+            someMethod();
+        } catch (Exception e) {
+            ex = new Exception("Wrapper");
+            addSuppressed(ex, e);
+            addSuppressed(e, ex);
+        }
+        
+        verify(ex);
+    }
+
+    @Test
+    public void suppressedWithCauseCircular() throws InvocationTargetException, IllegalAccessException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/tony19/logback/file/2370881413/issues/source?bid=3222618&fileBranchId=3222653#l248)